### PR TITLE
docs: update test section examples [ci skip]

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves []

### Description

Noticed that we encourage people to use `test-id` to query components, while `@testing-library/react-native` has different suggestions on querying components. Updated our docs to reflect to the best practices of the community 😊

I will also update the example links during the refineathon :) 

read more info here 👇
[How should I query?](https://callstack.github.io/react-native-testing-library/docs/how-should-i-query/)

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- updated how to test docs - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by pvinis** [comment]
do we have a good example to link underneath?

**Comment by gkartalis** [comment]
Great point, will find some examples and update the links, was thinking to update the links during the refineathon but rather sooner than later

File: EXAMPLES.md [hunk 1]
```diff
@@ -14,12 +14,13 @@ Links:
...
-We try to use `testID` to find elements.
+- We use `@testing-library/react-native` and our helper `renderWithWrappersTL`.
+- We refer to this guide 👉 [How should I query?](https://callstack.github.io/react-native-testing-library/docs/how-should-i-query/) from `@testing-library/react-native` for querying components
 
-Links:
-
-- [Dialog.tests.tsx](src/palette/elements/Dialog/Dialog.tests.tsx)
+Example Links:
+- [Search.tests.tsx](src/app/Scenes/Search/Search.tests.tsx)
+- [CustomSizeInputs.tests.tsx](src/app/Components/ArtworkFilter/Filters/CustomSizeInputs.tests.tsx)
```
Comment (pvinis) [inline EXAMPLES.md:22]: this is a nice one

**Review by pvinis — APPROVED** [review]
👍 nice

**Comment by patrinoua** [comment]
Thanks for showing our docs some love 🎉

**Pull request merged**